types/codecs/index.d.ts

```diff
@@ -0,0 +1,54 @@
+// Type definitions for codecs 2.2
+// Project: https://github.com/mafintosh/codecs
+// Definitions by: Martin Heidegger <https://github.com/martinheidegger>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+declare namespace codecs {
+    type JsonObject = { [Key in string]?: JsonValue };
+    interface JsonArray extends Array<JsonValue> {}
+    type JsonValue = string | number | boolean | null | JsonObject | JsonArray;
+
+    interface BaseCodec<InType = any, OutType = InType> {
+        encode(input: InType): Buffer;
+        decode(input: Uint8Array): OutType;
```

Comment:
I'm probably missing something obvious, but I'm not sure why the parameter type isn't `Buffer`.

Reply:
The project guarantees to return a Buffer instance when encoding but accepts a Uint8Array when decoding.